app/code/Magento/ImportExport/Controller/Adminhtml/Import/Download.php

```diff
@@ -68,22 +79,20 @@ public function __construct(
      */
     public function execute()
     {
-        $fileName = $this->getRequest()->getParam('filename') . '.csv';
-        $moduleDir = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, self::SAMPLE_FILES_MODULE);
-        $fileAbsolutePath = $moduleDir . '/Files/Sample/' . $fileName;
-        $directoryRead = $this->readFactory->create($moduleDir);
-        $filePath = $directoryRead->getRelativePath($fileAbsolutePath);
+        $entityName = $this->getRequest()->getParam('filename');
 
-        if (!$directoryRead->isFile($filePath)) {
+        try {
+            $fileContents = $this->sampleFileProvider->getFileContents($entityName);
```

Comment:
Trusting unverified input from the Request may be unsafe

Reply:
Done